language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java

```diff
@@ -142,6 +147,10 @@ public List<CodeAction> getCodeActions(Diagnostic diagnostic,
                 }
             }
 
+            if (checkExprNodeFinder.containCheckExprNode()) {
```

Comment:
Why do we need to check if the function body has `check` expressions? Instead, we should simply check if the function return type has an `error` member. If yes, we have to union the new return type with the error. WDYT?

Reply:
The issue here is we need to add `error` return type for possible return types if there is a `check` expression node in the function body. There can be cases where user specified `error` return and there are no any `error` returns in the function body.